pkg/fanal/secret/scanner.go

```diff
@@ -511,9 +514,16 @@ func findLocation(start, end int, content []byte) (int, int, types.Code, string)
 	rawLines := lines[codeStart:codeEnd]
 	var foundFirst bool
 	for i, rawLine := range rawLines {
-		strRawLine := string(rawLine)
 		realLine := codeStart + i
 		inCause := realLine >= startLineNum && realLine <= endLineNum
+
+		var strRawLine string
+		if len(rawLine) > maxLineLength {
+			strRawLine = lo.Ternary(inCause, matchLine, string(rawLine[:maxLineLength]))
+		} else {
+			strRawLine = string(rawLine)
+		}
```

Comment:
Looks like 100 characters is enough.
1. For lines before and after secret - we don't need such a big line.
2. For secret line - if length of secret with secret > 100 - we use `30 + secret + 20 characters`:
https://github.com/aquasecurity/trivy/blob/36ea01128f8e0cd84602a6c1d2c6e5f055ed795b/pkg/fanal/secret/scanner.go#L501-L504

> Now 2000 is a length for backward compatibility with RSA keys.

IIUC you want to include full RSA key to result (`-----BEGIN RSA PRIVATE KEY-----` and `-----END RSA PRIVATE KEY-----`), but I think this is not necessary.
But when we use 2000, we can see case when secret length is 10, but the user sees 1900 unneeded characters.